## Write a Python function called sqrt2 
- that calculates and prints to the screen the square root of 2 to 100 decimal places. 

Your code should not depend on any module from the standard library 1 or otherwise. 
You should research the task first and include references and a description of your algorithm.



Because the square root of 2 is an irrational number, as illustrated in the following reference [1], it ensures it is a peculiarity when it comes to using the value in calculations. 


[1] Proof: Square root of 2 is Irrational; Khan Academy; https://www.khanacademy.org/math/algebra/x2f8bb11595b61c86:irrational-numbers/x2f8bb11595b61c86:proofs-concerning-irrational-numbers/v/proof-that-square-root-of-2-is-irrational

Calculating the square root of 2 on my calculator, either on my phone, scientific calculator or on this computer, produc=es similar results. The result is always a number of varying lengths that resembles the following:
1.4142135623730950488016887242097 - to 31 decimal places from my computer.

In order to use Python to output the squre root of 2 to 100 decimal places, we need to look deeper at what constitutes the makeup of a square root. 

What exactly is a  square of a number? It is a number multiplied by itself. The square of 2 is 4 i.e. 2 X 2. The square of 9 is 81 i.e. 9 X 9. And likewise the square root is just one of those numbers we have multiplied by itself in order to get the result we require. 
That being said, the question we are to answer is "what number do we hveto multiply by itself to get 2? We know this is not an easy task, but what might be an easy task is to approximate the multiplication digit by digit. 

Lets look at multiplying 1 by 1 - too easy, the answer is 1. Now lets try 1.1 to see what we get?


In [1]:
x = 1.1 * 1.1

In [2]:
print("%.2f" % x)
x

1.21


1.2100000000000002

So the product of 1.1 is 1.21 - but our interest lies with 2, adn 1.21 is a  ways off. From our earlier calculation we could see that he square root of 2 was 1.414....... so we will start there. 



In [3]:
y = 1.414 * 1.414
print("%.3f" % y)
y

1.999


1.9993959999999997

This is as expected but if we were to use an algorithm that tested each of the digit locations, to see if the resulting multiplication resulted in the answer exceeding 2 then we can tweak it to be below 2 for each calculation. As an example lets look at 1.415:

In [4]:
z = 1.415 * 1.415
print("%.3f" % z)
z

2.002


2.002225

As expected, the result is above 2 which is not good. So we continue knowing that the first digits are 1.414. But to ensure the algorithm is fast we would want to check to see where within the 10 digits does the next number lie? Is it 1.4147 or i it 1.4142? We won;t concern ourselves with the speed of the algorithm for now, but if we get time to alter it before final submission then we can make improvements. For now we will run through the digits 0 to 9 in ascending order to get the closest fit and use hat nmber in our calculaton before proceeding to the next. 

However, after a simple check on alternate ways of expressing the swuare root of something, I might have an easier way of figuring this out. Checking this [2] website shows that the square root of something can be expressed as another number. For instance the square root of 2 can also be expressed as 2 to the power of a half! This was a mathematical fact that escaped me as I have not had to cover this topic in many years, so this is a bonus for me, and also from researching [3], it appears there is a built in function cintained within the standard library that should lend us an answer : pow(x,y) - x to the power of y.
[2] Rational Exponents; Monterey Institute; http://www.montereyinstitute.org/courses/DevelopmentalMath/COURSE_TEXT2_RESOURCE/U16_L1_T3_text_final.html
[3] The Python Standard Library, Built in types; Python.Org, https://docs.python.org/3/library/stdtypes.html#truth-value-testing

In [37]:
a = 2
b = 0.5
#c = (pow(a, b))
c
d = str(pow(a,b))
print("%.99f" % (pow(a,b)))
d
#c = c.as_integer_ratio()




1.414213562373095145474621858738828450441360473632812500000000000000000000000000000000000000000000000


'1.4142135623730951'

In [34]:
d = str(pow(a,b))
#print("%.99f" % c)
d
#c = c.as_integer_ratio()

'1.4142135623730951'